src/content/reference/react/useId.md

```diff
@@ -302,3 +302,40 @@ input { margin: 5px; }
 ```
 
 </Sandpack>
+
+### useId in server rendering {/*useid-in-server-rendering*/}
+
+Choose a unique id prefix and pass it into the server options and client options. `useId` will return the same id string on server side and client side. The following example selects `react-app1` as the id prefix. 
+
+```js
+import { useId } from 'react';
+
+function App() {
+  const id = useId();
+  // ...
+
+```
+
+```js
+/**
+ * server side
+ */
+
+import ReactServer from 'react-dom/server';
+
+const html = ReactServer.renderToString(<App />, { identifierPrefix: 'react-app1' });
```

Comment:
We should not use legacy APIs here. We can use `renderToPipeableStream` instead.